addons/l10n_in_pos/static/src/overrides/models/pos_order.js

```diff
@@ -21,85 +25,49 @@ patch(PosOrder.prototype, {
-                const all_taxes = line.compute_all(
-                    product_taxes,
-                    price_unit,
-                    quantity,
-                    line.pos.currency.rounding
-                );
-                //It is require to pre-compute GST Rate as we may have same hsn with different gst rates
-                const gstRate = formatFloat(this._computeL10nInGSTRate(all_taxes.taxes), {
-                    decimals: 2,
-                    trailingZeros: false,
-                });
-                const groupKey = "".concat(hsnCode, "-", gstRate);
-                if (!hsnSummary[groupKey]) {
-                    hsnSummary[groupKey] = {
-                        l10n_in_hsn_code: hsnCode,
-                        gst_rate: gstRate,
-                        SGST: { rate: 0, amount: 0 },
-                        CGST: { rate: 0, amount: 0 },
-                        IGST: { rate: 0, amount: 0 },
-                        CESS: { amount: 0 },
-                    };
-                }
-                const l10nInTaxKeyByTagRef = this._getl10nInTaxKeyByTagRef();
-                all_taxes.taxes.forEach((tax) => {
-                    tax.repartition_line_ids.forEach((line) => {
-                        line.tag_ids.forEach((tag) => {
-                            const tagRef = accountTagByXmlRefId[tag.id];
-                            const factor_percent = line.factor_percent;
-                            const taxKey = l10nInTaxKeyByTagRef[tagRef];
-                            switch (taxKey) {
-                                case "CGST":
-                                case "SGST":
-                                case "IGST":
-                                    hsnSummary[groupKey][taxKey]["amount"] +=
-                                        (factor_percent * tax.amount) / 100;
-                                    hsnSummary[groupKey][taxKey]["rate"] =
-                                        (factor_percent * tax.tax_rate) / 100;
-                                    break;
-                                case "CESS":
-                                    hsnSummary[groupKey][taxKey]["amount"] +=
-                                        (factor_percent * tax.amount) / 100;
-                                    break;
-                            }
-                        });
-                    });
-                });
+            if (!hsnCode) {
+                return;
             }
+
+            let taxes = line.tax_ids || line.product.taxes_id;
+            if (fiscalPosition) {
+                taxes = getTaxesAfterFiscalPosition(taxes, this.fiscal_position_id, this.models);
+            }
+
+            const priceUnit = line.get_unit_price();
+            baseLines.push({
+                l10n_in_hsn_code: hsnCode,
+                price_unit: priceUnit,
+                quantity: line.get_quantity(),
+                uom: null,
+                taxes: getTaxesValues(
```

Comment:
```suggestion
                tax_value_list: getTaxesValues(
```

https://github.com/odoo/odoo/pull/158522